perfkitbenchmarker/benchmarks/scimark2_benchmark.py

```diff
@@ -0,0 +1,265 @@
+    FFT             Mflops:  1568.64    (N=1024)
+    SOR             Mflops:  1039.98    (100 x 100)
+    MonteCarlo:     Mflops:   497.64
+    Sparse matmult  Mflops:  1974.39    (N=1000, nz=5000)
+    LU              Mflops:  2899.56    (M=100, N=100)
+
+  Sample Results (Java version):
+
+    SciMark 2.0a
+
+    Composite Score: 1731.4467627163242
+    FFT (1024): 996.9938397943672
+    SOR (100x100):   1333.5328291027124
+    Monte Carlo : 724.5221517116782
+    Sparse matmult (N=1000, nz=5000): 1488.18620413327
+    LU (100x100): 4113.998788839592
+
+    java.vendor: Oracle Corporation
+    java.version: 1.7.0_75
+    os.arch: amd64
+    os.name: Linux
+    os.version: 3.16.0-25-generic
+
+  Args:
+    results: SciMark2 result.
+
+  Returns:
+    A list of sample.Sample objects.
+  """
+  RESULT_START_C = '** SciMark2 Numeric Benchmark'
+  RESULT_START_JAVA = 'SciMark 2.0a'
+
+  SCORE_REGEX = r'\n(Composite Score):\s+(\d+\.\d+)'
+  RESULT_REGEX_C = r'\n(.+?)\s+Mflops:\s+(\d+\.\d+)(\s+\(.+?\))?'
+  RESULT_REGEX_JAVA = r'\n([^:]+):\s+(\d+\.\d+)'
+  PLATFORM_REGEX = r'\n(\w+\.\w+):\s+(.*)'
+
+  def FindBenchStart(results, start_index=0):
+    bench_version = 'Unknown'
+    index = results.find(RESULT_START_C, start_index)
+    if index != -1:
+      bench_version = 'C'
+    else:
+      index = results.find(RESULT_START_JAVA, start_index)
+      if index != -1:
+        bench_version = 'Java'
+    return index, bench_version
+
+  def ExtractPlatform(result, bench_version, clear=True):
+    metadata = dict()
+    if bench_version == 'C':
+      pass
+    elif bench_version == 'Java':
+      matches = re.finditer(PLATFORM_REGEX, result)
+      for m in matches:
+        metadata.update({m.group(1): m.group(2)})
+        if clear:
+          result = result[:m.start()] + ' ' * (m.end() - m.start()) + \
+              result[m.end():]
+    return metadata, result
+
+  def ExtractScore(result, bench_version, clear=True):
+    m = re.search(SCORE_REGEX, result)
+    label = m.group(1)
+    score = float(m.group(2))
+    if clear:
+      result = result[:m.start()] + ' ' * (m.end() - m.start()) + \
+          result[m.end():]
+    return score, label, result
+
+  def ExtractResults(result, bench_version):
+    datapoints = []
+    if bench_version == 'C':
+      match = regex_util.ExtractAllMatches(RESULT_REGEX_C, result)
+      for groups in match:
+        metric = '{0} {1}'.format(groups[0].strip(), groups[2].strip())
+        metric = metric.strip().strip(':')
+        value = float(groups[1])
+        datapoints.append((metric, value))
+    elif bench_version == 'Java':
+      match = regex_util.ExtractAllMatches(RESULT_REGEX_JAVA, result)
+      for groups in match:
+        datapoints.append((groups[0].strip(), float(groups[1])))
+    return datapoints
+
+  samples = []
+  start_index, bench_version = FindBenchStart(results)
+  while start_index != -1:
+    next_start, next_bench = FindBenchStart(results, start_index + 1)
```

Comment:
Alternatively, to avoid code duplication:

    start_index = -1
    while True:
        start_index, bench_version = FindBenchStart(results, start_index + 1)
        if start_index == -1:
            break